Slim/Schema.pm

```diff
@@ -3293,6 +3300,22 @@ sub canFulltextSearch {
 	return $canFulltextSearch;
 }
 
+sub _workRequired {
+	my $genres = shift;
+	return 1 if !$prefs->get('worksScan');
+	return 0 if $prefs->get('worksScan') == 2;
+	my $genreMatch = 0;
+	my %scanGenres = map {$_ => 1} split(/,\s*/, uc($prefs->get('myClassicalGenres')));
+	my @trackGenres = Slim::Music::Info::splitTag($genres);
+	foreach (@trackGenres) {
+		if ( $scanGenres{uc($_)} ) {
+			$genreMatch = 1;
+			last;
+		}
+	}
```

Comment:
With the above suggestion this could become as simple as:

```perl
my $genreMatch = grep { isClassicalGenre($_) } Slim::Music::Info::splitTag($genres);
```

Reply:
Would this unnecessarily carry on looping through the array after we've found a match?

Reply:
That's correct. But in this case we would rarely have more than one element anyway, and hardly ever more than two or three. I'd consider that potential overhead negligible.